synapse/handlers/oidc.py

```diff
@@ -957,6 +957,11 @@ async def grandfather_existing_users() -> Optional[str]:
                 # and attempt to match it.
                 attributes = await oidc_response_to_user_attributes(failures=0)
 
+                if attributes.localpart is None:
+                    # If no localpart is returned then we will generate one, so
+                    # there is no need to search for existing users.
+                    return None
```

Comment:
I guess this previously worked by there hopefully not being an `@None:foo` user?

Reply:
Yup, I believe so